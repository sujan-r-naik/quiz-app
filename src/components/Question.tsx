import { Box, Typography } from "@mui/material";
import Divider from "@mui/material/Divider";
import { col, row } from "../constants/FlexStyles";

interface IQuestion {
  q: string;
  index: number;
}

const fixQuestion = (str: string) => {
  const newString = str.replace(
    "Match the following JavaScript function with their description: ",
    ""
  );
  const words = newString.split(" | ");
  const colA = words.filter((x, i) => {
    if (i % 2 === 0) return x;
  });
  const colB = words.filter((x, i) => {
    if (i % 2 !== 0) return x;
  });
  return [colA, colB];
};

const Question = (props: IQuestion) => {
  return (
    <>
      {props.index === 4 ? (
        <>
          <p style={{ textAlign: "center", marginBottom: "7px" }}>
            Match the Following (eg: 1: a, 2: b)
          </p>
          <Box sx={{ width: "100%", mb: "2rem", ...row }}>
            <Box sx={{ ...col, width: "40%" }}>
              {fixQuestion(props.q)[0].map((x, i) => (
                <Typography role={"match-qa"} key={i} sx={{ fontSize: "0.8rem" }}>
                  {x}
                </Typography>
              ))}
            </Box>
            <Box sx={{ ...col, width: "50%" }}>
              {fixQuestion(props.q)[1].map((x, i) => (
                <Typography role={"match-qb"} key={i} sx={{ fontSize: "0.8rem" }}>
                  {x}
                </Typography>
              ))}
            </Box>
          </Box>
        </>
      ) : (
        <Box sx={{ width: "100%", mb: "2rem" }}>
          <Typography role={"normal-q"} sx={{ fontSize: "1rem" }}>{props.q}</Typography>
        </Box>
      )}
    </>
  );
};

export default Question;
