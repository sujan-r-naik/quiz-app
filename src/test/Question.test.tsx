import "@testing-library/jest-dom/extend-expect";
import { screen, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Question from "../components/Question";
import renderer from "react-test-renderer";

test("Snapshot of NumberNav component", () => {
  const comp = renderer.create(
    <BrowserRouter>
      <Question q={"Question"} index={4} />
    </BrowserRouter>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("In Quiz,", () => {
  test("does the sub components of question work", () => {
    render(
      <BrowserRouter>
        <Question q={"Question"} index={4} />
      </BrowserRouter>
    );

    let matchqa = screen.getByRole("match-qa");
    expect(matchqa).toBeInTheDocument();
  });
});
