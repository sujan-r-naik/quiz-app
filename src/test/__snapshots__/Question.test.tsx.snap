// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of NumberNav component 1`] = `
Array [
  <p
    style={
      Object {
        "marginBottom": "7px",
        "textAlign": "center",
      }
    }
  >
    Match the Following (eg: 1: a, 2: b)
  </p>,
  <div
    className="MuiBox-root css-m7eb2g"
  >
    <div
      className="MuiBox-root css-14jbbht"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-r9rcpu-MuiTypography-root"
        role="match-qa"
      >
        Question
      </p>
    </div>
    <div
      className="MuiBox-root css-13ce5na"
    />
  </div>,
]
`;
